modules/apps/bean-portlet/bean-portlet-cdi-extension/src/main/java/com/liferay/bean/portlet/cdi/extension/internal/InjectionTargetWrapper.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.bean.portlet.cdi.extension.internal;
+
+import java.util.Set;
+
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.enterprise.inject.spi.InjectionTarget;
+
+/**
+ * @author Dante Wang
+ */
+public class InjectionTargetWrapper<T> implements InjectionTarget<T> {
+
+	public InjectionTargetWrapper(InjectionTarget<T> injectionTarget) {
+		_injectionTarget = injectionTarget;
+	}
+
+	@Override
+	public void dispose(T instance) {
+		_injectionTarget.dispose(instance);
+	}
+
+	@Override
+	public Set<InjectionPoint> getInjectionPoints() {
+		return _injectionTarget.getInjectionPoints();
+	}
+
+	@Override
+	public void inject(T instance, CreationalContext<T> ctx) {
+		_injectionTarget.inject(instance, ctx);
+	}
+
+	@Override
+	public void postConstruct(T instance) {
+		_injectionTarget.postConstruct(instance);
+	}
+
+	@Override
+	public void preDestroy(T instance) {
+		_injectionTarget.preDestroy(instance);
+
+		_callback.run();
```

Comment:
Because this method does a little bit more than what a plain wrapper method would do, I recommend that you rename the class to `PreDestroyInjectionTarget`. Or you could have an `InjectionTargetWrapper` class that simply wraps and delegates (nothing more) and then have a `PreDestroyInjectionTarget` class that extends it and overrides only the `preDestroy` method.